src/Validation/src/RuntimeValidatableTypeInfoResolver.cs

```diff
@@ -0,0 +1,390 @@
+        }
+
+        info = CreateValidatableTypeInfo(type, new HashSet<Type>());
+        _cache.TryAdd(type, info);
+        return info is not null;
+    }
+
+    /// <summary>
+    /// Attempts to get validatable parameter information for the specified parameter.
+    /// </summary>
+    /// <param name="parameterInfo">The parameter to get validation information for.</param>
+    /// <param name="validatableInfo">When this method returns, contains the validatable parameter information if found; otherwise, <see langword="null"/>.</param>
+    /// <returns><see langword="true"/> if validatable parameter information was found; otherwise, <see langword="false"/>.</returns>
+    /// <remarks>
+    /// This implementation always returns <see langword="false"/> as parameter resolution is handled by <see cref="RuntimeValidatableParameterInfoResolver"/>.
+    /// </remarks>
+    public bool TryGetValidatableParameterInfo(
+        ParameterInfo parameterInfo,
+        [NotNullWhen(true)] out IValidatableInfo? validatableInfo)
+    {
+        validatableInfo = null;
+        return false;
+    }
+
+    private static RuntimeValidatableTypeInfo? CreateValidatableTypeInfo(Type type, HashSet<Type> visitedTypes)
+    {
+        // Prevent infinite recursion by tracking visited types
+        if (!visitedTypes.Add(type))
+        {
+            return null;
+        }
+
+        try
+        {
+            // Skip types that don't need validation (same logic as parameter resolver)
+            if (!IsClassForType(type))
+            {
+                return null;
+            }
+
+            // Get validation attributes applied to the type
+            var typeValidationAttributes = type
+                .GetCustomAttributes<ValidationAttribute>()
+                .ToArray();
+
+            // Get all public instance properties
+            var properties = type.GetProperties(BindingFlags.Public | BindingFlags.Instance);
+            var validatableProperties = new List<RuntimeValidatablePropertyInfo>();
+
+            foreach (var property in properties)
+            {
+                // Skip properties without setters (read-only properties) for normal classes
+                if (!property.CanWrite && !IsRecordType(type))
+                {
+                    continue;
+                }
+
+                // Skip properties marked with [FromService] or [FromKeyedService] attributes
+                if (HasFromServiceAttributes(property.GetCustomAttributes()))
+                {
+                    continue;
+                }
+
+                // Get validation attributes for this property
+                var propertyValidationAttributes = property
+                    .GetCustomAttributes<ValidationAttribute>()
+                    .ToArray();
+
+                // For record types, also check constructor parameters for validation attributes
+                if (propertyValidationAttributes.Length == 0 && IsRecordType(type))
```

Comment:
Seems like this would skip the record ctor if it also had parameters?

```
record struct H([Range(1, 10)] int A)
{
    [Required]
    public bool B { get; set; }
}
```